### PR TITLE
ad env var and group config for entity_name conjunction

### PR DESCRIPTION
**Pull request by AlvaroVega**
related https://github.com/telefonicaid/iotagent-node-lib/issues/944

The new configuration option can be set by agent or group  service (no sense for device)

### CODE REVIEW

File: lib/commonConfig.js [hunk 1]
```diff
@@ -155,6 +155,7 @@ function processEnvironmentVariables() {
         'IOTA_MULTI_CORE',
```
Comment (fgalan) [inline lib/commonConfig.js:155]: CNR entry needed?

File: lib/commonConfig.js [hunk 2]
```diff
@@ -155,6 +155,7 @@ function processEnvironmentVariables() {
...
         'IOTA_DEFAULT_EXPRESSION_LANGUAGE',
         'IOTA_RELAX_TEMPLATE_VALIDATION',
+        'IOTA_DEFAULT_ENTITY_NAME_CONJUNCTION',
```
Comment (fgalan) [inline lib/commonConfig.js:158]: It should be included in documentation (env var and corresponding config.js parameter)

File: lib/commonConfig.js (hunk 1 above)
Comment (fgalan) [reply lib/commonConfig.js:155]: ...mentioning defaultEntityNameConjunction and its associated env var

File: lib/commonConfig.js (hunk 2 above)
Comment (AlvaroVega) [reply lib/commonConfig.js:158]: fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/949/commits/832a13516e1e539d81cb52eafacfce51e3df8c0d

File: lib/commonConfig.js (hunk 1 above)
Comment (AlvaroVega) [reply lib/commonConfig.js:155]: fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/949/commits/1435a44f80e8d3fe10c672bdc214952f4c31229b

File: doc/installationguide.md [hunk 3]
```diff
@@ -278,6 +278,8 @@ used for the same purpose. For instance:
     standard, but the final decision has yet been confirmed), take into account it could change
 -   **explicitAttrs**: if this flag is activated, only provisioned attributes will be processed to Context Broker. This
     flag is overwritten by `explicitAttrs` flag in group or device provision.
+-   **defaultEntityNameConjunction**: the default conjunction string used to compose a default entity_name when is not 
+    provided at device provisioning time. This value is overwritten by `defaultEntityNameConjunction` in group provision.
```
Comment (fgalan) [inline doc/installationguide.md:282]: > ... This value is overwritten by `defaultEntityNameConjunction` in group provision

Two things:

1. Just to confirm the field can be used only in group provision but not in device provision
2. The new field `defaultEntityNameConjunction` should be added to the group provision documentation (not sure in which .md is... I refer to the table we have with all possible fiels).

File: doc/installationguide.md [hunk 4]
```diff
@@ -345,6 +347,7 @@ overrides.
 | IOTA_FALLBACK_PATH               | `fallbackPath`                  |
 | IOTA_DEFAULT_EXPRESSION_LANGUAGE | `defaultExpressionLanguage`     |
 | IOTA_EXPLICIT_ATTRS              | `explicitAttrs`                 |
+| IOTA_DEFAULT_ENTITY_NAME_CONJUNCTION | `efaultEntityNameConjunction` |
```
Comment (fgalan) [inline doc/installationguide.md:350]:
```suggestion
| IOTA_DEFAULT_ENTITY_NAME_CONJUNCTION | `defaultEntityNameConjunction` |
```

File: doc/installationguide.md (hunk 3 above)
Comment (AlvaroVega) [reply doc/installationguide.md]: better explained by https://github.com/telefonicaid/iotagent-node-lib/pull/949/commits/d870b94d738aa99436852ca730f40bf4becb4f1c

Comment (fgalan) [reply doc/installationguide.md]: >   ... (not sure in which .md is... I refer to the table we have with all possible fields)

I mean this: https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#service-group-model

Comment (AlvaroVega) [reply doc/installationguide.md]: fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/949/commits/2321e665aef37d5824bba499168319002dd595b4

File: CHANGES_NEXT_RELEASE [hunk 5]
```diff
@@ -1,3 +1,4 @@
+Add env var and config for default entity_name conjunction (#944)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
Add: defaultEntityNameConjunction config (env var IOTA_DEFAULT_ENTITY_NAME_CONJUNCTION) and configuration group API field for default entity_name conjunction (#944)
```

**Review by fgalan — APPROVED** [review]
LGTM

1st one in merging queue (after CI e2e gets stable again)

**Comment by fgalan** [comment]
Now that CI e2e is happy again, this PR can be merged

**Pull request merged**